### PR TITLE
Scala doc improvment

### PR DESCRIPTION
**Pull request by zarinfam**
Fix how to access deadLetters from system in actor.

### CODE REVIEW

File: akka-docs/rst/java/code/docs/io/japi/IODocTest.java [hunk 1]
```diff
@@ -90,6 +90,10 @@ public void onReceive(Object msg) throws Exception {
     
     final InetSocketAddress remote;
     final ActorRef listener;
+    
+    public static Props props(InetSocketAddress remote, ActorRef listener) {
+        return Props.create(TcpClientActor.class, remote, listener);
+    }
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/io/japi/IODocTest.java]: Does not seem related to described fix?

File: akka-docs/rst/scala/code/docs/actor/ActorDocSpec.scala [hunk 2]
```diff
@@ -454,7 +454,7 @@ class ActorDocSpec extends AkkaSpec(Map("akka.loglevel" -> "INFO")) {
       class WatchActor extends Actor {
         val child = context.actorOf(Props.empty, "child")
         context.watch(child) // <-- this is the only call needed for registration
-        var lastSender = system.deadLetters
+        var lastSender = context.system.deadLetters
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/actor/ActorDocSpec.scala]: Right, good catch.

File: akka-docs/rst/java/code/docs/io/japi/IODocTest.java (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/java/code/docs/io/japi/IODocTest.java]: Doesn't hurt in any case though... Thanks

**Comment by ktoso** [comment]
LGTM, however it seems the branch has merge conflicts, please rebase¿

**Comment by ktoso** [comment]
OK TO TEST

File: akka-docs/rst/java/code/docs/io/japi/IODocTest.java (hunk 1 above)
Comment (zarinfam) [inline akka-docs/rst/java/code/docs/io/japi/IODocTest.java]: It was an old contribution and i do not know why it is here!

**Comment by zarinfam** [comment]
Can delete this pull request and send another one again?

**Comment by ktoso** [comment]
Sure, thanks!
